src/specific/s_input.c

```diff
@@ -467,6 +809,130 @@ static int16_t S_Input_JoyAxis(SDL_GameControllerAxis axis)
     return 0;
 }
 
+static INPUT_STATE S_Input_GetControllerState(
+    INPUT_STATE state, INPUT_LAYOUT cntlr_layout_num)
+{
+    // clang-format off
+    for (int role = 0; role < INPUT_ROLE_NUMBER_OF; role++) {
+        CONTROLLER_MAP assigned = m_ControllerLayout[cntlr_layout_num][role];
+        int16_t btn_state = 0;
+
+        if (assigned.type == BT_BUTTON) {
+            btn_state = S_Input_JoyBtn(assigned.bind.button);
+        } else {
+            btn_state = S_Input_JoyAxis(assigned.bind.axis) == assigned.axis_dir;
+        }
+
+        switch(role) {
+        case INPUT_ROLE_UP:
```

Comment:
I'd prefer to see this compressed to individual lines and safe-guarded with clang-format off. Otherwise we don't need to disable the formatting.

Reply:
It does with one tiny remark. In general after big block like this:
```
keyword {
    continuous lines
    continuous lines
}
```
I like to have a single blank line. The reason is to cater for IDEs that have "jump to the next blank line" and for aesthetics.